extensions/wikia/EditPageLayout/js/editor/AceEditor.js

```diff
@@ -86,18 +86,39 @@ define(
 	 * Init submit action
 	 */
 	function initSubmit() {
-		$('#editform').submit(function(e) {
-			var hiddenInput = ace.getInput().val(ace.getContent());
+		$('#editform').submit(function (e) {
+			var hiddenInput = ace.getInput().val(ace.getContent()),
+				$editForm = $(this);
 
 			disableBeforeUnload = true;
 
-			$(this).unbind('submit').append(hiddenInput).submit();
-
 			e.preventDefault();
+
+			/* Allow TemplateClassification extension to break submit */
+			if (win.enableTemplateClassificationEditorPlugin === true) {
+				require(['TemplateClassificationModalForce'], function forceTemplateClassificationModal(tcForce) {
+					if (tcForce.forceType() === true) {
+						return;
+					} else {
+						finishSubmit($editForm, hiddenInput);
+					}
+				});
+			} else {
+				finishSubmit($editForm, hiddenInput);
+			}
 		});
 	}
 
 	/**
+	 * Submit form with hidden field
+	 * @param {jQuery} $editForm
+	 * @param {jQuery} hiddenInput
+	 */
+	function finishSubmit($editForm, hiddenInput) {
+		$editForm.unbind('submit').append(hiddenInput).submit();
```

Comment:
If hiddenInput is a jQuery object, I would consider adding a "$" sign before it's name, by convention, as you do with editForm.